Add `car put-block` command

parkan opened the pull request:
- write counterpart to get-block
- can set root on first block write (optional)

this is handy if you need to package up some blocks from an IPFS node (among other cases); `ipfs dag export` wants to walk the whole DAG, which doesn't work if you only need a shallow traversal

use like:
```
ipfs block get $CID1 | car put-block foo.car --set-root --version=2 --codec=dag-pb
ipfs block get $CID2 | car put-block foo.car --version=2 --codec=dag-pb
```

I tried to match the semantics to how the other write commands work

willscott commented on cmd/car/car.go:
```diff
@@ -231,6 +231,29 @@ func main1() int {
+					&cli.StringFlag{
+						Name:    "codec",
+						Aliases: []string{"c"},
+						Usage:   "Codec to use for CID generation (raw, dag-pb, dag-cbor, dag-json)",
```
in the use case examples you provided, i wonder if there is an 'auto-detect' that's worthwhile in trying to auto 'upgrade' to the standard json/cbor/pb when data can be parsed as such - or if there's a way to get this from the cid rather than potentially loosing it?

parkan replied:
for the _very specific_ usecase yes, we could autodetect it, but I feel like requiring the cid as well as contents of the block will be annoying in the more general case; we could accept a `--cid` parameter, with either codec or cid needing to be set

parkan commented on cmd/car/car:
ooops, finger slip

parkan commented on cmd/car/put.go:
```diff
@@ -0,0 +1,134 @@
+	// Finalize the blockstore
+	if err := bs.Finalize(); err != nil {
+		return fmt.Errorf("failed to finalize CAR file: %w", err)
+	}
```
my design goal was to minimize my code surface and use the provided APIs as much as possible; actually I should check whether v1 appends work at all (and if they don't I'll do a straight append, yeah)

parkan commented:
ok, I've addressed the concerns:
- removed binary checkin oopsie (should it be gitignored?)
- v1 cars we do naive/cheap append
- accept `--cid` param, if passed we parse codec out of that and verify that recomputed value matches; otherwise `--codec` must be passed
